packages/renderer/src/lib/ContainerList.svelte

```diff
@@ -40,18 +40,48 @@ import type { ContextUI } from './context/context';
 import Button from './ui/Button.svelte';
 import StateChange from './ui/StateChange.svelte';
 import SolidPodIcon from './images/SolidPodIcon.svelte';
+import Tab from './ui/Tab.svelte';
+import { findMatchInLeaves } from '../stores/search-util';
+import { Filter } from './container/filter';
 
 const containerUtils = new ContainerUtils();
 let openChoiceModal = false;
 let enginesList: EngineInfoUI[];
 
 // groups of containers that will be displayed
 let containerGroups: ContainerGroupInfoUI[] = [];
-let viewContributions: ViewInfoUI[] = [];
-let globalContext: ContextUI;
-let containersInfo: ContainerInfo[] = [];
-export let searchTerm = '';
-$: searchPattern.set(searchTerm);
+
+let tab: string;
+
+export let filter = '';
+let runningFilter: boolean;
+let stoppedFilter: boolean;
+
+$: filterObj = new Filter(decodeURI(filter));
+
+$: checkFilter(filterObj.rawFilter);
+
+function checkFilter(_: string): void {
+  if (filterObj.isRunning() && !runningFilter) {
+    router.goto(`/containers/list/running?filter=${filterObj.setState(true, false)}`);
```

Comment:
reading all these setState(true/false) is not really meaningful without looking at the function itself. I think it could be made more readable by using some specific function (`setRunningState`), or as you have the information already in the url "running/stopped" you can make a function that create it (`createRunningURL(/containers/list/)`) .